sparse/core.py

```diff
@@ -589,70 +589,103 @@ def __neg__(self):
                    self.sorted)
 
     def __sub__(self, other):
-        return self.elemwise_binary(operator.sub, other)
+        return self.elemwise(operator.sub, other)
 
     def __rsub__(self, other):
         return -(self - other)
 
     def __mul__(self, other):
-        return self.elemwise_binary(operator.mul, other)
+        return self.elemwise(operator.mul, other)
 
     __rmul__ = __mul__
 
     def __truediv__(self, other):
-        return self.elemwise_binary(operator.truediv, other)
+        return self.elemwise(operator.truediv, other)
 
     def __floordiv__(self, other):
-        return self.elemwise_binary(operator.floordiv, other)
+        return self.elemwise(operator.floordiv, other)
 
     __div__ = __truediv__
 
     def __pow__(self, other):
-        return self.elemwise_binary(operator.pow, other)
+        return self.elemwise(operator.pow, other)
 
     def __and__(self, other):
-        return self.elemwise_binary(operator.and_, other)
+        return self.elemwise(operator.and_, other)
 
     def __xor__(self, other):
-        return self.elemwise_binary(operator.xor, other)
+        return self.elemwise(operator.xor, other)
 
     def __or__(self, other):
-        return self.elemwise_binary(operator.or_, other)
+        return self.elemwise(operator.or_, other)
 
     def __gt__(self, other):
-        return self.elemwise_binary(operator.gt, other)
+        return self.elemwise(operator.gt, other)
 
     def __ge__(self, other):
-        return self.elemwise_binary(operator.ge, other)
+        return self.elemwise(operator.ge, other)
 
     def __lt__(self, other):
-        return self.elemwise_binary(operator.lt, other)
+        return self.elemwise(operator.lt, other)
 
     def __le__(self, other):
-        return self.elemwise_binary(operator.le, other)
+        return self.elemwise(operator.le, other)
 
     def __eq__(self, other):
-        return self.elemwise_binary(operator.eq, other)
+        return self.elemwise(operator.eq, other)
 
     def __ne__(self, other):
-        return self.elemwise_binary(operator.ne, other)
+        return self.elemwise(operator.ne, other)
 
     def elemwise(self, func, *args, **kwargs):
+        """
```

Comment:
I needed your opinion here. Should we keep it this way and assume `other=args[0]` or should we add an `other=None` argument for explicitly selecting between binary and unary functions?

Reply:
The first is more compact but the second can help when, for example, the first input to your function itself is supposed to be `COO` or `spmatrix`.

Reply:
Without thinking about this too much I'm inclined to leave this as `def elemwise(self, func, *args, **kwargs)`.  I would expect `len(args)` to determine between elemwise_unary and elemwise_binary as you've done below.

Reply:
I'm not sure I understand the second case

Reply:
Suppose you have an inherently unary function that accepts `COO` or `spmatrix` as its second positional argument, instead of it being an operand to a binary function. I agree, it's a corner use case, but it is more explicit.

Reply:
In case you think this won't ever be needed (and I mostly agree), this should be good to merge, if you haven't got any other comments.

Reply:
I wouldn't worry about this for now.  Short term I might encourage users to use kwrags for this.

Reply:
Fair enough. I overlooked that kwargs can map onto normal args in Python!